Sources/xcresultparser/CoberturaCoverageConverter.swift

```diff
@@ -0,0 +1,202 @@
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import Foundation
+import XCResultKit
+
+public class CoberturaCoverageConverter: CoverageConverter {
+    public override func xmlString(quiet: Bool) throws -> String {
+        guard
+            let dtdUrl = URL(string: "http://cobertura.sourceforge.net/xml/coverage-04.dtd"),
+            let dtd = try? XMLDTD(contentsOf: dtdUrl) else {
+                fatalError("DTD could not be constructed")
+        }
+
+        dtd.name = "coverage"
+        dtd.systemID = "http://cobertura.sourceforge.net/xml/coverage-04.dtd"
+
+        let rootElement = self.makeRootElement()
+
+        let doc = XMLDocument(rootElement: rootElement)
+        doc.version = "1.0"
+        doc.dtd = dtd
+        doc.documentContentKind = XMLDocument.ContentKind.xml
+
+        let sourceElement = XMLElement(name: "sources")
+        rootElement.addChild(sourceElement)
+        sourceElement.addChild(XMLElement(name: "source", stringValue: projectRoot.isEmpty ? "." : projectRoot))
+
+        let packagesElement = XMLElement(name: "packages")
+        rootElement.addChild(packagesElement)
+
+        let fileInfoSemaphore = DispatchSemaphore(value: 1)
+        let files = try coverageFileList()
+        var fileInfo: [FileInfo] = []
+        
+        // since we need to invoke xccov for each file, it takes pretty much time
+        // so we invoke it in parallel on 8 threads, that speeds up things considerably
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 8 //Deadlock if this is = 1
+        queue.qualityOfService = .userInitiated
+        for file in files {
+            guard !file.isEmpty else { continue }
+            if !quiet {
+                writeToStdError("Coverage for: \(file)\n")
+            }
+            let op = BlockOperation { [self] in
+                do {
+                    let file = try fileCoverage(for: file, relativeTo: projectRoot)
```

Comment:
I find it a bit misleading to use 'file' here as variable name, as it is the already used by the variable, which is the current for-loop value. I named it 'coverage' in my code for that reason. Any reason you renamed it to use the same variable name as the loop variable?

Reply:
Good catch, I didn't realize I was shadowing a variable here, I will rename it.